src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java

```diff
@@ -1,50 +1,46 @@
 package de.wejago.hichi2influx.repository;
 
 import com.influxdb.client.InfluxDBClient;
-import com.influxdb.client.WriteApiBlocking;
-import com.influxdb.client.domain.WritePrecision;
+import com.influxdb.client.WriteApi;
 import com.influxdb.client.write.Point;
 import de.wejago.hichi2influx.config.InfluxDBConfig;
-import de.wejago.hichi2influx.dto.SensorEntry;
-import java.time.Instant;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Repository;
 
 @Repository
 @RequiredArgsConstructor
+@Slf4j
+@EnableScheduling
 public class InfluxDbRepository {
     private final InfluxDBConfig influxDBConfig;
-
     private InfluxDBClient influxDBClient;
+    private Queue<Point> measurementPoints = new LinkedBlockingQueue<>();
 
-    public Point writePoint(SensorEntry sensorEntry) {
-        updateConnectionIfNeeded();
-        WriteApiBlocking writeApi = influxDBClient.getWriteApiBlocking();
-
-        Point point = generateMeasurementPoint(sensorEntry);
-        writeApi.writePoint(point);
-
-        return point;
+    public void writePoint(Point point) {
+        measurementPoints.add(point);
     }
 
-    private void updateConnectionIfNeeded() {
-        //set the connection only if it is empty of not active
-        if(influxDBClient == null || !influxDBClient.ping()) {
-            influxDBClient = influxDBConfig.dbConnection();
+    @Scheduled(fixedRate = 30000)
+    public void persistPointsToInflux() {
+        log.info("persistPointsToInflux() - every 30 secs");
+        influxDBClient = influxDBConfig.updateConnectionIfNeeded();
+        int numberOfPointsWritten = 0;
+
+        if(influxDBClient.ping()) {
+            log.info("InfluxDbRepository.writePoint() with successful ping()");
+            while (!measurementPoints.isEmpty()) {
+                Point point = measurementPoints.poll();
+                WriteApi writeApi = influxDBClient.makeWriteApi();
```

Comment:
Do not create a new Api on every write, that's why it currently dies. See Nasko's but report.

Does the Api also have the ping? Then the DBConfig could return the api instead of the client.

Reply:
I used the @Bean trick to create it. And from what I read by default it is singleton so should be fine.